instant-xml-macros/src/de.rs

```diff
@@ -242,13 +241,49 @@ impl Deserializer {
         def_prefix: Option<String>,
         field_namespace: Option<String>,
     ) {
+        let mut is_option_type = false;
         let field_var = field.ident.as_ref().unwrap();
         let field_var_str = field_var.to_string();
         let const_field_var_str = Ident::new(&field_var_str.to_uppercase(), Span::call_site());
         let field_type = match &field.ty {
             syn::Type::Path(v) => match v.path.get_ident() {
                 Some(ident) => ident.into_token_stream(),
-                None => (&v.path.segments.first().expect("Struct name").ident).into_token_stream(),
+                None => {
+                    if v.path.segments.is_empty() {
+                        panic!("Wrong declaration");
+                    };
+
+                    let type_params = v.path.segments.first().expect("Struct name");
+                    match type_params.ident.to_string().as_str() {
+                        "Option" => match &type_params.arguments {
+                            syn::PathArguments::AngleBracketed(params) => {
+                                is_option_type = true;
+                                let mut temp = quote!(Option<);
+                                match params.args.first().unwrap() {
+                                    syn::GenericArgument::Type(syn::Type::Path(v)) => {
+                                        match v.path.get_ident() {
+                                            Some(ident) => temp.extend(ident.into_token_stream()),
+                                            None => temp.extend(
+                                                (&v.path
+                                                    .segments
+                                                    .first()
+                                                    .expect("Struct name")
+                                                    .ident)
+                                                    .into_token_stream(),
+                                            ),
+                                        }
+                                    }
+                                    _ => panic!("Wrong data"),
+                                }
+
+                                temp.extend(quote!(>));
+                                temp
+                            }
+                            _ => panic!("Wrong data"),
+                        },
```

Comment:
I definitely think this is wrong, and we should find another way to do this. I've been looking at the serde output and source code but haven't quite figured it out yet.

Reply:
This looks ugly, because I need to manually remove lifetime from the definition, otherwise it would be much easier.
Something like: `v.path.segments.first().expect("Struct name").into_tokens_stream()`. Serde is using this function heavily, maybe it will be helpful: https://github.com/serde-rs/serde/blob/f52d134c14f6904010d0a59107987b050b36b811/serde_derive/src/de.rs#L3120
I will take a look. But I will focus first on the rest of PRs.